src/xpcc/driver/position/vl53l0.hpp

```diff
@@ -0,0 +1,531 @@
+ */
+template < typename I2cMaster >
+class Vl53l0 : public vl53l0, public xpcc::I2cDevice< I2cMaster, 5 >
+{
+public:
+	/// Constructor, requires an vl53l0::Data object, sets address to default of 0x29
+	Vl53l0(Data &data, uint8_t address=0x29);
+
+	/// Ping the I2C device.
+	/// This overwrites the build-in ping method of I2cDevice with a read of the model number.
+	xpcc::ResumableResult<bool>
+	ping();
+
+	xpcc::ResumableResult<bool>
+	initialize();
+
+	/// Set a new I2C address (< 128) for this device.
+	/// The address is not permanent and must be set again after every device boot.
+	xpcc::ResumableResult<bool>
+	setDeviceAddress(uint8_t address);
+
+	/// Reads the distance and buffers the result
+	inline xpcc::ResumableResult<bool>
+	readDistance()
+	{ return readSensor(); }
+
+	inline RangeErrorCode
+	getRangeError()
+	{ return data.error; }
+
+	template <typename T>
+	xpcc::ResumableResult<bool>
+	updateRegister(Register reg, T setMask, T clearMask = T(0xff))
+	{
+		return updateControlRegister(reg, Control_t(setMask), Control_t(clearMask));
+	}
+
+	/// the data object for this sensor.
+	inline Data&
+	getData()
+	{ return data; }
+
+	/// Set the maximum measurement time.
+	///
+	/// Increasing this value will improve the accuracy.
+	/// The default is ~30ms. ST recommends 2000000us = 200ms for high precision.
+	///
+	/// useful range of values: ~20000us - 2000000us (20ms - 2s)
+	xpcc::ResumableResult<bool>
+	setMaxMeasurementTime(uint32_t timeUs);
+
+	/// Get the configured maximum measurement time.
+	inline uint32_t
+	getMaxMeasurementTime()
+	{ return measurementTimeUs; }
+protected:
+	/// @cond
+	/// read multiple 8bit values from subsequent registers
+	xpcc::ResumableResult<bool>
+	read(Register reg, uint8_t *buffer, uint8_t length);
+	/// @endcond
+
+	/// @cond
+	/// read an 8bit value from a register
+	xpcc::ResumableResult<bool> xpcc_always_inline
+	read(Register reg, uint8_t& value)
+	{ return read(reg, &value, 1); }
+	/// @endcond
+
+	/// @cond
+	/// write to a control register
+	xpcc::ResumableResult<bool> xpcc_always_inline
+	write(Register reg, Control_t value)
+	{ return write(reg, value.value); }
+	/// @endcond
+
+	/// @cond
+	/// write an 8bit value to a register
+	xpcc::ResumableResult<bool>
+	write(Register reg, uint8_t value);
+	/// @endcond
+
+	/// @cond
+	/// write a 16bit value to 2 subsequent registers in big endian format
+	xpcc::ResumableResult<bool>
+	writeUInt16(Register reg, uint16_t value);
+	/// @endcond
+
+	/// @cond
+	xpcc::ResumableResult<bool>
+	updateControlRegister(Register reg, Control_t setMask, Control_t clearMask = Control_t(0xff));
+	/// @endcond
+
+	/// @cond
+	/// periodically read an 8 bit value from a register until the supplied
+	/// predicate returns true
+	///
+	/// the predicate must take a uint8_t argument and return a bool
+	template < typename Predicate >
+	xpcc::ResumableResult<bool>
+	poll(Register reg, Predicate pred, uint16_t timeoutMs = 20, uint16_t stepMs = 4);
```

Comment:
> won't allow you to use a lambda with a capture
I assume because the closure needs to be stored somewhere and a C pointer just stores a pointer, correct?